Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Parameters/SpeckleBlockDefinitionWrapper.ModelObjects.cs

```diff
@@ -44,5 +39,53 @@ private bool CastToModelObject<T>(ref T target)
 
     return false;
   }
+
+  private bool CastFromModelInstanceDefinition(ModelInstanceDefinition modelInstanceDef)
+  {
+    var doc = RhinoDoc.ActiveDoc;
+    var instanceDef = doc?.InstanceDefinitions.Find(modelInstanceDef.Name);
+    if (instanceDef == null)
+    {
+      // Rhino → Model → Model Block Definition passthrough component returns type ModelInstanceDefinition
+      // .Objects of a ModelInstanceDefinition returns ModelObjects
+      // ModelObject.Geometry is internal and cannot be accessed directly.
+      // Only way to get geometry from a ModelObject is through RhinoDoc.Objects.FindId(), which only works for baked objects.
+      // Unbaked Grasshopper geometry cannot be processed through the ModelObject workflow until we get a public geometry accessor 😓
+      // ⚠️ So if user defines a Model Block Definition in Grasshopper with Grasshopper (unbaked) geometry, we're stuck.
+      // That's why we're intercepting this case early → if the instanceDef == null don't go further
+      throw new InvalidOperationException(
+        $"Block definition '{modelInstanceDef.Name}' not found in Rhino document. Please bake the definition first or use Speckle Block Definition components instead."
+      );
+    }
+
+    var objects = new List<SpeckleObjectWrapper>();
+
+    var modelObjects = modelInstanceDef.Objects ?? Array.Empty<ModelObject>();
+
+    foreach (var modelObj in modelObjects)
+    {
+      var objWrapperGoo = new SpeckleObjectWrapperGoo();
+      if (objWrapperGoo.CastFrom(modelObj))
+      {
+        objects.Add(objWrapperGoo.Value);
+      }
+    }
+
+    Value = new SpeckleBlockDefinitionWrapper()
+    {
+      Base = new InstanceDefinitionProxy
+      {
+        name = modelInstanceDef.Name,
+        applicationId = modelInstanceDef.Id?.ToString() ?? Guid.NewGuid().ToString(),
+        objects = objects.Select(o => o.ApplicationId ?? Guid.NewGuid().ToString()).ToList(),
+        maxDepth = 1
```

Comment:
No changes yet, but a not that this max depth will need to be stored on the wrapper class and calculated, when you get to supporting nested blocks